contractcourt/commit_sweep_resolver.go

```diff
@@ -191,14 +209,31 @@ func (c *commitSweepResolver) Resolve() (ContractResolver, error) {
 	c.currentReport.MaturityHeight = unlockHeight
 	c.reportLock.Unlock()
 
-	// If there is a csv delay, we'll wait for that.
-	if c.commitResolution.MaturityDelay > 0 {
-		c.log.Debugf("waiting for csv lock to expire at height %v",
-			unlockHeight)
+	// If there is a csv/cltv lock, we'll wait for that.
+	if c.commitResolution.MaturityDelay > 0 || c.hasCLTV() {
+		// Determine what height we should wait until for the locks to
+		// expire.
+		var waitHeight uint32
+		switch {
+		// If we have both a csv and cltv lock, we'll need to look at
+		// both and see which expires later.
+		case c.commitResolution.MaturityDelay > 0 && c.hasCLTV():
```

Comment:
Ah I see it's handled down here, could just as well do it up there w/ a max statement.

Reply:
Using `math.Max` now, but we still need another check to determine whether we can broadcast a block before the timelock expires, which is only valid when there's no CLTV or the CSV is after the CLTV.